src/main/java/org/zalando/nakadi/service/AuthorizationValidator.java

```diff
@@ -155,7 +155,9 @@ public void authorizeStreamRead(final EventType eventType) throws AccessDeniedEx
         final Resource resource = new EventTypeResource(eventType.getName(), eventType.getAuthorization());
         try {
             if (!authorizationService.isAuthorized(AuthorizationService.Operation.READ, resource)) {
-                throw new AccessDeniedException(AuthorizationService.Operation.READ, resource);
+                if (!adminService.hasAllDataAccess(AuthorizationService.Operation.READ)) {
+                    throw new AccessDeniedException(AuthorizationService.Operation.READ, resource);
+                }
             }
         } catch (final PluginException e) {
             throw new ServiceTemporarilyUnavailableException("Error calling authorization plugin", e);
```

Comment:
Maybe it is normal in Java, just learning, again,
Why you Throw and Catch and Throw again. Instead one Throw, maybe complex like:
```
... e =new AccessDeniedException(AuthorizationService.Operation.READ, resource);
throw new ServiceTemporarilyUnavailableException("Error calling authorization plugin", e);
```

Reply:
Here we catch PluginException, which is thrown from inside a plugin (so we may not necessarily know how it behaves). We catch it, then throw ServiceTemporarilyUnavailableException, so we don't expose implementation details to the caller. Or at least that's how I see it

Reply:
Ah, I see, you are catching possible errors from authorizationService and adminService.